### ISSUE TITLE
[RunBook]: Migrate lambda function to gravitron2 processor

### ISSUE THREAD

**Issue by amit-chandak-unskript** [unskript/Awesome-CloudOps-Automation#539]
### RunBook Name

Migrate lambda function to gravitron2 processor

### runbook Inputs

region, list of lambda function names

### runBook Actions

#537

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by shloka-bhalgat-unskript** [comment]
@amit-chandak-unskript The command to update the configuration architecture for a non graviton2 process is `aws lambda update-function-code --function-name <function-name> --zip-file`

We need a new zip file of the arm64 compatible code. Once that is updated the new runtime architecture also get's updated. Since we don't have a way to upload files for a lego, I think we can contain the scope of this runbook to **posting a slack message of all the lambda functions that don't use the graviton2 processor**.

Also the reference article that you linked is not really migrating the lambda function, they are simply creating a new one using create-lambda and specifying ` -architectures arm64`

I found this too - https://repost.aws/questions/QUe69VPTSHS3mVnJj9KL-12w/bug-cannot-change-lambda-architecture-without-providing-a-new-zip-file-in-the-management-console

**Comment by shloka-bhalgat-unskript** [comment]
That's correct @amit-chandak-unskript , I tried it for the  case where binaries were not needed to be recompiled. We could simply update the runtime configuration for architecture type.

But, there is no way to do it. Documentation  for [update-lambda-configuration](https://docs.aws.amazon.com/cli/latest/reference/lambda/update-function-configuration.html)

The `--architectures ` is not a valid flag for this api even when I specify the runtime (like python3.7)
<img width="1474" alt="Screenshot 2023-04-26 at 6 33 19 PM" src="https://user-images.githubusercontent.com/110628398/234583371-3e47727f-5803-44f1-90f2-f82ec03cd1d4.png">

As per the article , we can do it via console but there isn't a way to **update** the architecture configuration for the function. They have simply created a new one...

_When using the AWS CLI to create a Lambda function, specify the --architectures option. If you do not specify the architecture, the default value is x86-64. For example, to create an arm64 function, specify--architectures arm64._

```
aws lambda create-function \
    --function-name MyArmFunction \
    --runtime nodejs14.x \
    --architectures arm64 \
    --memory-size 512 \
    --zip-file fileb://MyArmFunction.zip \
    --handler lambda.handler \
    --role arn:aws:iam::123456789012:role/service-role/MyArmFunction-role
```

**Comment by amit-chandak-unskript** [comment]
https://docs.aws.amazon.com/cli/latest/reference/lambda/update-function-code.html

Use this instead @shloka-bhalgat-unskript
